src/realtime/price.updates.ts

```diff
@@ -209,35 +198,40 @@ export class PriceUpdates {
         const that = this;
 
         setTimeout(() => {
-            if (!isEmpty(that.subscribersWithTicker)) {
-                that.subscribersWithTicker.forEach((symbolTicker) => {
-                    log(`cancelMktData ${symbolTicker.symbol} tickerId=${symbolTicker.tickerId}`);
-                    that.ib.cancelMktData(symbolTicker.tickerId);
-                });
+            for (const symbolTicker of Object.values(this.tickerIdToData)) {
+                log(`cancelMktData ${symbolTicker.symbol} tickerId=${symbolTicker.tickerId}`);
+                that.ib.cancelMktData(symbolTicker.tickerId);
             }
-            return;
+            this.tickerIdToData = {};
```

Comment:
don't these 2 lines reset all other tickerIdToData?

have you tested to subscribe more than one symbol,

Reply:
Will test.

Reply:
I've pushed a new commit.  Yes, the lines completely reset `this.tickerIdToData` and `this.keyToTickerId`; isn't that the intended purpose of `unsubscribeAllSymbols()`?  However, I pushed a commit that moves it to the top of the function so that the reset is immediate instead of waiting for the timeout.  Let me know if you see a problem with the current patch.